models/Schematic_DataModel.php

```diff
@@ -40,14 +40,21 @@ protected function defineAttributes()
      * Populate data model from yaml.
      *
      * @param string $yaml
+     * @param string $overrideYaml
      *
      * @return Schematic_DataModel
      */
-    public static function fromYaml($yaml)
+    public static function fromYaml($yaml, $overrideYaml)
     {
         $data = Yaml::parse($yaml);
+        $overrideData = Yaml::parse($overrideYaml);
+        if ($overrideData != null) {
+            $mergedData = array_replace_recursive($data, $overrideData);
```

Comment:
it's array_replace_recursive

http://php.net/manual/en/function.array-replace-recursive.php

When the value in array1 is scalar, it will be replaced by the value in array2, may it be scalar or array. When the value in array1 and array2 are both arrays, array_replace_recursive() will replace their respective value recursively.